bat-utils/lib/hapi-server.js

```diff
@@ -66,194 +74,112 @@ const Server = async (options, runtime) => {
     })
   }
 
-  if (graylist.addresses) {
-    graylist.authorizedAddrs = []
-    graylist.authorizedBlocks = []
-
-    graylist.addresses.forEach((entry) => {
-      if ((entry.indexOf('/') === -1) && (entry.split('.').length === 4)) return graylist.authorizedAddrs.push(entry)
-
-      graylist.authorizedBlocks.push(new Netmask(entry))
-    })
-  }
-
-  if (runtime.prometheus) {
-    await new Promise((resolve, reject) => {
-      try {
-        server.register(runtime.prometheus.plugin(), (err) => {
-          if (err) {
-            reject(err)
-            throw err
-          } else {
-            resolve()
-          }
-        })
-      } catch (e) {
-        debug(e)
-        reject(e)
-      }
-    })
-  }
-
-  await new Promise((resolve, reject) => {
-    server.register([
+  const { prometheus } = runtime
+  const plugins = [].concat(
+    prometheus ? [
+      prometheus.plugin()
+    ] : [],
+    [
       bell,
-      asyncHandler,
       authBearerToken,
-      authCookie,
-      whitelist,
+      hapiCookie,
+      {
+        plugin: whitelist.plugin
+      },
       inert,
       vision,
+      rateLimiter(runtime),
+      swagger()
+    ], process.env.NODE_ENV === 'production' ? [
```

Comment:
I'd also enforce HTTPs for `staging`

Reply:
somewhat confusingly we also set `NODE_ENV` to production on staging :)